tests: counter: counter_callback.single_shot_alarm_notop build failure for frdm_k64f

### Describe the bug

- frdm_k64f

99fd6d4f4e172e713e0614bb52e3497f802f4fef is the first bad commit
commit 99fd6d4f4e172e713e0614bb52e3497f802f4fef
Author: Tim Pambor <tim.pambor@codewrights.de>
Date:   Tue Oct 14 19:12:14 2025 +0200

    ztest: Add validation of zassert strings
    
    Add __printf_like attribute to ztest assertion functions
    zassert, zassume, zexpect to validate format strings and arguments
    at compile time.
    
    Signed-off-by: Tim Pambor <tim.pambor@codewrights.de>

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b frdm_k64f tests/drivers/counter
```

### Relevant log output

```shell
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/ztest/include/zephyr/ztest_assert.h:
197:40: note: in expansion of macro 'COND_CODE_1' 197 | _zassert_va(cond, default_msg, COND_CODE_1(__VA_OPT__(1), (__VA_ARGS__), (NULL))) | ^~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/subsys/testsuite/ztest/include/zephyr/ztest_assert.h:
324:34: note: in expansion of macro 'zassert' 324 | #define zassert_equal(a, b, ...) zassert((a) == (b), #a " not equal to " #b, ##__VA_ARGS__) | ^~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/counter/maxim_ds3231_api/src/test_counter.c:
400:9: note: in expansion of macro 'zassert_equal' 400 | zassert_equal(-ENOTSUP, err, | ^~~~~~~~~~~~~/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/tests/drivers/counter/maxim_ds3231_api/src/test_counter.c:
401:62: note: format string is defined here 401 | "%s: Counter failed to set top value: %d", dev->name); | ~^ | | | intcc1: all warnings being treated as errorsninja: build stopped: subcommand failed.
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: ubuntu
- toolchain: Zephyr
- version: v4.2.0-6134-g6b8fcaeecf95

### Additional Context

_No response_

--- a/tests/drivers/counter/maxim_ds3231_api/src/test_counter.c
+++ b/tests/drivers/counter/maxim_ds3231_api/src/test_counter.c
@@ -398,7 +398,7 @@ void test_multiple_alarms_instance(const struct device *dev)
 
 	err = counter_set_top_value(dev, &top_cfg);
 	zassert_equal(-ENOTSUP, err,
-		      "%s: Counter failed to set top value: %d", dev->name);
+		      "%s: Counter failed to set top value: %d", dev->name, err);
 
 	k_sleep(K_USEC(3 * (uint32_t)counter_ticks_to_us(dev, alarm_cfg.ticks)));
 

